server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -198,6 +203,9 @@ public MetadataCreateIndexService(
 
         // Task is onboarded for throttling, it will get retried from associated TransportClusterManagerNodeAction.
         createIndexTaskKey = clusterService.registerClusterManagerTask(ClusterManagerTaskKeys.CREATE_INDEX_KEY, true);
+        remoteStoreBlobPathResolver = isRemoteStoreAttributePresent(settings)
```

Comment:
we should use `isRemoteDataAttributePresent` instead since we only need it for translog and segments .

Reply:
Rebased from main and changing this.